pkg/server/file.go

```diff
@@ -129,6 +132,11 @@ func (t *TranslateServiceServer) UploadTranslationFile(
 		// Original translation is not affected, changes will not affect other translations - update incoming translation.
 		all = model.Translations{*translation}
 	case translation.Original && origIdx != -1:
+		if translation.Language != all[origIdx].Language {
+			return nil, status.Errorf(
+				codes.AlreadyExists, "original translation already exists for service: '%s'", params.serviceID)
```

Comment:
`Already exists` applies to the same entity. The better error is `conflict`.

Reply:
hmm, couldn't find conflict status code for grpc.

Reply:
in that case, we can use INVALID_ARGUMENT